elastic/kibana
# [Custom threshold] Improve the custom threshold rule request/response logging

Related to #187271

## Summary

As mentioned in this [PR](https://github.com/elastic/kibana/pull/187225), when logging a JSON stringified string for debugging or tracing, we need to wrap it in a function. This PR does this for the custom threshold rule and removes extra request/response logging, relying on alerting framework logging for that purpose.

![image](https://github.com/user-attachments/assets/c1fa8bb2-e895-4aae-a8da-b74fe5a8ca1e)

Before, we were able to have a similar log using `plugins.observability` and `plugins.alerting.observability.rules.custom_threshold` configs (as shown above), but now we only have `plugins.alerting.observability.rules.custom_threshold` which can be enabled by either of the following configs:

```
logging:
  loggers:
    - name: plugins.alerting
      level: debug
    - name: plugins.alerting.observability.rules.custom_threshold
      level: trace
```

Thanks @dgieselaar for bringing this to our attention!

### How to test
- Enable the trace logging config as mentioned above
- Create a custom threshold rule and check the server logs; you should be able to see the logs for the request and response of the rule execution.
